apps/chat/src/contexts/SearchContext.tsx

```diff
@@ -148,112 +141,106 @@ export const SearchContextProvider = ({ children }: Props) => {
     // Save to history.
     setHistory(addHistoryItem({ query: value, filter, language }, history));
 
-    // First call - only search results - should come back quicky while we wait for summarization
     setIsSearching(true);
     setIsSummarizing(true);
     setSelectedSearchResultPosition(undefined);
 
-    let initialSearchResponse;
+    const startTime = Date.now();
+    let resultsWithSnippets;
 
     try {
-      const startTime = Date.now();
-      initialSearchResponse = await sendSearchRequest({
-        filter,
-        queryValue: value,
-        rerank: rerank.isEnabled,
-        rerankNumResults: rerank.numResults,
-        rerankerId: rerank.id,
-        rerankDiversityBias: rerank.diversityBias,
-        hybridNumWords: hybrid.numWords,
-        hybridLambdaLong: hybrid.lambdaLong,
-        hybridLambdaShort: hybrid.lambdaShort,
-        customerId: search.customerId!,
-        corpusId: search.corpusId!,
-        endpoint: search.endpoint!,
-        apiKey: search.apiKey!
-      });
-      const totalTime = Date.now() - startTime;
-
-      // If we send multiple requests in rapid succession, we only want to
-      // display the results of the most recent request.
-      if (searchId === searchCount) {
-        setIsSearching(false);
-        setSearchTime(totalTime);
-        setSearchResponse(initialSearchResponse);
-
-        if (initialSearchResponse.response.length > 0) {
-          setSearchError(undefined);
-        } else {
-          setSearchError({
-            message: "There weren't any results for your search."
-          });
+      const onStreamEvent = (event: ApiV2.StreamEvent) => {
+        switch (event.type) {
+          case "requestError":
+          case "genericError":
+            setSearchError({
+              message: "Error sending the query request"
+            });
+            break;
+
+          case "error":
+            setSummarizationError({ message: event.messages.join(", ") });
+            break;
+
+          case "searchResults":
+            setIsSearching(false);
+            setSearchTime(Date.now() - startTime);
+            resultsWithSnippets = event.searchResults.map((result: SearchResult) => {
```

Comment:
We do this a couple of times in this PR. The `SearchResult` type returned with stream query `searchResults` events doesn't have a snippet. So when we receive them, we parse the snippet and tack it on, forming the `SearchResultWithSnippet` type, which is a simple extension of `SearchResult`.